### ISSUE TITLE
Use Redis-Proto as the javascript parser

### ISSUE THREAD

**Issue by steelbrain** [NodeRedis/node_redis#754]
https://github.com/mranney/node_redis/blob/master/lib/parser/javascript.js vs https://github.com/steelbrain/Redis-Proto/blob/master/Source/Main.js
Less code = faster, easier to maintain

**Comment by kadishmal** [comment]
Less code doesn't mean faster. Can you point to any benchmark stating it's faster? Easier to maintain, perhaps.

**Comment by BridgeAR** [comment]
@steelbrain did you run some benchmarks on the implemented js parser and yours?

**Comment by steelbrain** [comment]
@BridgeAR not yet, but I do have plans for that

**Comment by BridgeAR** [comment]
Ping @steelbrain

**Comment by steelbrain** [comment]
@BridgeAR I am sorry but I haven't been able to gather enough time to benchmark them both. I'll make this top priority for tomorrow :)

**Comment by steelbrain** [comment]
@BridgeAR I have done some performance related tweaks to `redis-proto` and have also released it's 1.0. I am not much experienced with these benchmark libraries for node.js, I have this code in which `node_redis`'s parser doesn't seem to be working. I presume you have experience in these, would you like to fix it and post the results?

```js
'use strict'
const RedisProto = require('redis-proto')
const Node_Redis = require('./node_redis').Parser // Path to a local copy of js parser
const Benchmark = require('benchmark')

const Content = RedisProto.Encode('Hello Dolly')

new Benchmark.Suite()
.add('RedisProto', function() {
  RedisProto.Decode(Content)
})
.add('NodeRedis', function(deferred) {
  var decoder = new Node_Redis
  decoder.send_reply = function() {
    deferred.resolve()
  }
  decoder.execute(Content)
}, {'defer': true})
.on('cycle', function(event) {
  console.log(String(event.target));
})
.on('complete', function() {
  console.log('Fastest is ' + this.filter('fastest').pluck('name'));
})
.on('error', function(e) {
  console.log(e)
})
.run();

```

**Comment by BridgeAR** [comment]
@steelbrain I'm sorry I do not have the time to do this. If you managed to write a benchmark and the numbers are looking good feel free to open a new issue.
